api/v1/gitrepository_types.go

```diff
@@ -38,6 +38,21 @@ const (
 	IncludeUnavailableCondition string = "IncludeUnavailable"
 )
 
+// GitVerificationMode specifies the verification mode for a Git repository.
+type GitVerificationMode string
+
+const (
+	// ModeHead implies that the HEAD of the Git repository (after it has been
+	// checked out to the required commit) should be verified.
+	ModeHead GitVerificationMode = "head"
+	// ModeTag implies that the tag object specified in the checkout configuration
+	// should be verified.
+	ModeTag GitVerificationMode = "tag"
+	// ModeTagAndHead implies that both the tag object and the commit it points
+	// to should be verified.
+	ModeTagAndHead GitVerificationMode = "tagAndHead"
```

Comment:
The idiomatic Kubernetes API values would be `HEAD`, `Tag` and `TagAndHead`, see also e.g. https://kubernetes.io/docs/concepts/containers/images/#image-pull-policy

Reply:
we can't allow `Head`/`HEAD` as that'd result in a breaking change, since the enum's only variant right now is `head` (unless kubernetes validation is case insensitive, in which case this change is fine).

Reply:
We can continue to have `head` as an accepted enum value, while internally transforming it into `HEAD`.

Which would be the case if my validation logic from below is applied, in which `head` is not recognized and then automatically transformed into the `HEAD` default.